processor/resourcedetectionprocessor/internal/system/system.go

```diff
@@ -89,6 +96,9 @@ func (d *Detector) Detect(ctx context.Context) (resource pcommon.Resource, schem
 			}
 			d.rb.SetHostArch(hostArch)
 			d.rb.SetOsDescription(osDescription)
+			if len(cpuInfo) > 0 {
+				setHostCPUInfo(d, cpuInfo[0])
```

Comment:
I hadn't considered the possibility that different CPUs could, in principle have different vendor IDs/stepping/cache sizes. I think that should be extremely rare, but it feels like we should log what CPU we are taking the information from here.

It is something we should consider for the frequency metric though

Reply:
I had the same thoughts on this. Ideally we should use an attribute like `host.cpu.logical_number` or `system.cpu.logical_number` (from https://github.com/open-telemetry/semantic-conventions/pull/89#pullrequestreview-1621813178) to distinguish those. But I agree that it would be quite rare so the distinction would be somehow redundant and not storage efficient. If in the future we find the need for this we can extend.

I will add the log you suggested since it would be helpful for any potential debugging here.

And true that the frequency will need this distinction(the metric's number are different per cpu).